app/src/main/java/com/example/playlistmaker/search/ui/view/SearchFragment.kt

```diff
@@ -1,47 +1,51 @@
 package com.example.playlistmaker.search.ui.view
 
-import android.content.Context
+import android.content.Context.INPUT_METHOD_SERVICE
 import android.content.Intent
 import android.graphics.drawable.Drawable
 import android.os.Bundle
 import android.os.Handler
 import android.os.Looper
 import android.text.Editable
 import android.text.TextWatcher
+import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
 import android.view.inputmethod.EditorInfo
 import android.view.inputmethod.InputMethodManager
 import android.widget.*
-import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.resource.bitmap.CenterCrop
 import com.example.playlistmaker.R
-import com.example.playlistmaker.databinding.ActivitySearchBinding
-import com.example.playlistmaker.player.ui.view.AudioplayerActivity
+import com.example.playlistmaker.databinding.FragmentSearchBinding
 import com.example.playlistmaker.models.Track
+import com.example.playlistmaker.player.ui.view.AudioplayerActivity
 import com.example.playlistmaker.search.domain.models.SearchState
 import com.example.playlistmaker.search.ui.viewmodel.SearchViewModel
 import com.example.playlistmaker.utils.KEY_TRACK_ID
 import com.example.playlistmaker.utils.createJsonFromTrack
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class SearchActivity : AppCompatActivity() {
-
+class SearchFragment: Fragment() {
     private lateinit var searchEditTxt: EditText
     private lateinit var btnPlaceHolderUpdate: Button
     private lateinit var placeHolder: LinearLayout
     private lateinit var placeHolderImage: ImageView
     private lateinit var placeHolderMessage: TextView
     private lateinit var progressBar: ProgressBar
     private lateinit var trackRecyclerView: RecyclerView
+    private lateinit var btnClearEditTxt: ImageView
     private var textWatcher: TextWatcher? = null
     private var isClicked = true
     private val handler = Handler(Looper.getMainLooper())
     private val viewModel by viewModel<SearchViewModel>()
 
-    private lateinit var binding: ActivitySearchBinding
+    private var _binding: FragmentSearchBinding? = null
+    private val binding: FragmentSearchBinding
+        get() = _binding!!
 
     private val playlistAdapter = SearchAdapter(
         object : SearchAdapter.SearchClickListener {
```

Comment:
🍏 Можно упростить запись таким образом:
```suggestion
    private val playlistAdapter = SearchAdapter { track ->
        viewModel.addNewTrackToHistory(track)
        startActivity(track)
    }
```